magento/magento2
# Subscribe to Newsletter. Multistore. Issue#13258

### Description
This fix solves the problem of subscribe to the Newsletter in Multistore setup.
With this fix, the newsletter subscritions are related to the store. The newsletter subscritions are now linked to the email and the store ID, allowing the same user to subscribe the newsletter in store A and store B.

### Fixed Issues (if relevant)
1. magento/magento2#13258: Subscribe to Newsletter. Multistore.

### Manual testing scenarios
1. Create a multistore setup
2. Subscribe one email in store A
3. Subscribe the same email adress in Store B
4. Verify that the Newsletter list has two entries for the same email address but for different stores
5. Try to subscribe the same email adress on store A
6. Verify that you receive the error related with the email address already subscribed
7. Repeat the same steps, now with customer registration
8. On the customer menu you sould see the correct subscription status related with each store
9. Edit the subscripton on store A or B and see that you are changing the corresponding line on the newsletter subscription list.

### Contribution checklist
